stl/inc/atomic

```diff
@@ -475,14 +475,33 @@ bool __stdcall _Atomic_wait_compare_non_lock_free(
 
 #ifdef _WIN64
 inline bool __stdcall _Atomic_wait_compare_16_bytes(const void* _Storage, void* _Comparand, size_t, void*) noexcept {
+
+#ifdef _M_ARM64
+#ifndef __clang__
     const auto _Dest              = static_cast<long long*>(const_cast<void*>(_Storage));
     const auto _Cmp               = static_cast<const long long*>(_Comparand);
     alignas(16) long long _Tmp[2] = {_Cmp[0], _Cmp[1]};
-#ifdef _M_X64
-    return _STD_COMPARE_EXCHANGE_128(_Dest, _Tmp[1], _Tmp[0], _Tmp) != 0;
-#else // ^^^ _M_X64 / ARM64 vvv
+
     return _InterlockedCompareExchange128_nf(_Dest, _Tmp[1], _Tmp[0], _Tmp) != 0;
-#endif // ^^^ ARM64 ^^^
+#else // ^^^ !__clang__ / __clang__ vvv
+    __int128* _Dest           = static_cast<__int128*>(const_cast<void*>(_Storage));
+    alignas(16) __int128 _Tmp = static_cast<const __int128*>(_Comparand)[0];
+
+    do {
+        if (__builtin_arm_ldrex(_Dest) != _Tmp) {
+            return false;
+        }
+    } while (__builtin_arm_strex(_Tmp, _Dest));
+
+    return true;
+#endif // ^^^ __clang__ ^^^

```

Comment:
Since Clang has implemented `_InterlockedCompareExchange128_nf` upstream, I believe that this should be marked as `TRANSITION, Clang 12`.